test/std/algorithms/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -0,0 +1,103 @@
+// <algorithm>
+
+// template<ForwardIterator Iter, class T>
+//   constexpr bool      // constexpr after C++17
+//   binary_search(Iter first, Iter last, const T& value);
+
+#include "oneapi_std_test_config.h"
+#include "test_iterators.h"
+#include "checkData.h"
+
+#include <iostream>
+
+#ifdef USE_ONEAPI_STD
+#    include _ONEAPI_STD_TEST_HEADER(algorithm)
+#    include _ONEAPI_STD_TEST_HEADER(iterator)
+namespace s = oneapi_cpp_ns;
+#else
+#    include <algorithm>
+#    include <iterator>
+namespace s = std;
+#endif
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <class Iter, class T>
+bool
+test(Iter first, Iter last, const T& value, bool x)
+{
+    return (s::binary_search(first, last, value) == x);
+}
+
+class KernelBSearchTest1;
+class KernelBSearchTest2;
+class KernelBSearchTest3;
+class KernelBSearchTest4;
+
+template <typename Iter, typename KC>
+void
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    sycl::cl_bool ret = false;
+    sycl::range<1> numOfItems{1};
+    sycl::buffer<sycl::cl_bool, 1> buffer1(&ret, numOfItems);
+    const unsigned N = 1000;
+    const int M = 10;
+    int host_vbuf[N];
+    int x = 0;
+    for (size_t i = 0; i < N; ++i)
+    {
+        host_vbuf[i] = x;
+        if (++x == M)
+            x = 0;
+    }
+
+    std::sort(host_vbuf, host_vbuf + N);
+    sycl::range<1> host_buffer_sz{N};
+    sycl::buffer<sycl::cl_int, 1> host_data_buffer(host_vbuf, host_buffer_sz);
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        auto ret_access = buffer1.get_access<sycl_write>(cgh);
+        auto data_access = host_data_buffer.get_access<sycl_read>(cgh);
+        cgh.single_task<KC>([=]() {
```

Comment:
This is also from source test state.
I think out goal is simple to import tests into our repository.

Reply:
Mikhail, what complication do you see?
We test binary_search, yes - but over the data passed into the kernel as a buffer. That makes sense to me.
Or are you concerned about ret_access etc.? It is just the way to combine results of multiple checks with boolean AND, and return the value to the host.

Anyway, as Sergey mentioned, the primary goal for now is to integrate these tests into our repo. It's OK to point out some obvious and simple improvements, but making these tests "perfect" is not a goal.

Reply:
"the primary goal for now is to integrate these tests into our repo. "is to integrate these tests into our repo."
(from https://github.com/intel-innersource/applications.compilers.source.dpcpplib)

I would ask "why do we need that?" Why do we need such duplication of huge amount (hundreds files) of the tests (especially in "oneIntel" context). Why we cannot rely on "dpcpplib" test reports? What we should do if that huge amount test will have divergence between dpcpplib and oneDPL repo?

Of source, in case if dpcpplib repo "is died" - situation more or less clear.

Reply:
@akukanov could you please take a look at https://github.com/oneapi-src/oneDPL/pull/885#discussion_r1358096648 ?

Reply:
applications.compilers.source.dpcpplib is an old, de-facto abandoned internal repository for oneDPL, Except for these tests, all its content was already moved to our main repository. I prefer to have everything related to oneDPL project in a single place, instead of two.